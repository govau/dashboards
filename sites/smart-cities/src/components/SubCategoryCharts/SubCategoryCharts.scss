@import '~style/main';

.wrapper {
  background: $color-white;
  padding-top: 24px;
}

.container {
  @include container;
}

.heading {
  display: flex;
  flex-direction: column;
  padding: 16px;

  @include for-tablet-landscape-up {
    flex-direction: row;
    align-items: center;
    padding: 24px 32px;
  }
}

.headingText {
  font-size: 44px;
  font-weight: 300;
  line-height: 1.2;
  margin-bottom: 8px;

  @include for-tablet-landscape-up {
    margin-bottom: 0;
  }
}

.patternToggle {
  @include for-tablet-landscape-up {
    margin-left: auto;
  }
}

.chartGrid {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
}

.chartWrapper {
  width: 100%;

  @include for-tablet-portrait-up {
    width: calc(50% - 16px);
  }
}

.heroChartWrapper {
  width: 100%;
}

.iconWrapper {
  margin-right: 20px;
}
