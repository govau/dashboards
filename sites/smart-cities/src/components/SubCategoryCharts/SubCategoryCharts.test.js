import React from 'react';
import { shallow } from 'enzyme';
import SubCategoryCharts from './SubCategoryCharts';

jest.mock('../CityColumnChart/CityColumnChart', () => 'CityColumnChart');
jest.mock('../Icon/Icon', () => 'Icon');
jest.mock('../Toggle/Toggle', () => 'Toggle');

const setShowChartPatternsMock = jest.fn();

const defaultProps = {
  subCategory: {
    name: 'Buying houses',
    highlightColorDark: 'JOBS_600',
    summaryIndicatorIds: [],
    charts: [],
  },
  cities: [
    {
      name: 'Perth',
      indicators: {}
    },
    {
      name: 'Sydney',
      indicators: {}
    },
  ],
  colorName: 'jobs',
  heroIndicatorId: 'population',
  highlightColorDark: 'JOBS_600',
  setShowChartPatterns: setShowChartPatternsMock,
  showChartPatterns: false,
};

it('should match Snapshot', () => {
  const component = shallow(
    <SubCategoryCharts {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should show chart patterns when showChartPatterns is true', () => {
  const component = shallow(
    <SubCategoryCharts
      {...defaultProps}
      showChartPatterns={true}
    />
  );

  expect(component.find('svg').length).toBe(1);
  expect(component.find('pattern').length).toBe(4);
});

it('should not show chart patterns when showChartPatterns is false', () => {
  const component = shallow(
    <SubCategoryCharts
      {...defaultProps}
      showChartPatterns={false}
    />
  );

  expect(component.find('svg').length).toBe(0);
});

it('should toggle showPatterns when the toggle is clicked', () => {
  const component = shallow(
    <SubCategoryCharts {...defaultProps} />
  );

  component.find('.patternToggle').simulate('change');

  expect(setShowChartPatternsMock).toHaveBeenCalledTimes(1);
});
