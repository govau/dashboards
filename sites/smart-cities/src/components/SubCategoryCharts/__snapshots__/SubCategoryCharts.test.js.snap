// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"wrapper\\">
  <section className=\\"container\\" id=\\"sc-buying-houses\\">
    <header className=\\"heading\\" style={{...}}>
      <Icon className=\\"iconWrapper\\" size={70} icon={[undefined]} color=\\"JOBS_600\\" />
      <h1 className=\\"headingText\\">
        Buying houses
      </h1>
      <Toggle className=\\"patternToggle\\" id=\\"chart-pattern-checkbox-buying-houses\\" label=\\"Make charts more accessible\\" onChange={[Function]} checked={false} colorName=\\"jobs\\" />
    </header>
    <div className=\\"chartGrid\\" />
  </section>
</div>"
`;
